tests/server/query_params/test_filter.py

```diff
@@ -1,15 +1,20 @@
-"""Make sure response_fields is handled correctly"""
+"""Make sure filters are handled correctly"""
 import pytest
+from optimade.server.config import CONFIG, SupportedBackend
```

Comment:
Not sure what the problems that you're alluding to are; each pytest session is going to be testing one fixed backend so it shouldn't break anything. Importing inside each test will have the same side effect anyway, as the client will only be created once.

The different ways of handling it correspond to different outcomes: either needing to check for different error messages, or marking the tests as `xfail`s.

Reply:
I'm unsure about the import order, but I just remember there were issues with importing configuration settings and similar outside tests as opposed to inside. But if it works, it's fine, I guess.

Reply:
It works fine with this current layout, i.e. testing each backend indepedently. It would be nice if we could wrap a lot of these tests with some kind of parameterized context manager that can switch between the backends (this will also be more extensible if we add more backends) but that is beyond the scope of this PR